src/clojurians_log/db/queries.clj

```diff
@@ -63,16 +64,16 @@
              :in '[$ ?chan-name ?day]
              :where '[[?msg :message/day ?day]
                       [?msg :message/channel ?chan]
-                      [?msg :message/user ?user]
                       [?chan :channel/name ?chan-name]]}
             db
             chan-name
             day)
        (map first)
        ;; Remove all thread messages except for the thread parent
+       ;; and except for brodcast messages.
        ;; Note that thread parents do not have a :thread-ts value themselves
-       (remove #(if-let [thread-ts (:message/thread-ts %)]
-                  (not= thread-ts (:message/ts %))))
+       (remove #(when-let [thread-ts (:message/thread-ts %)] 
+                  (not (contains? % :message/thread-broadcast?))))
```

Comment:
This isn't very clear, I'm having a hard time figuring out if this is correct or not. You're no longer using `thread-ts` so you shouldn't be using `when-let`.

Let's start by rewriting this as

``` clojure
(remove #(and (:message/thread-ts %) (not (contains? % :message/thread-broadcast?))))
```

So remove everything that is a thread message, unless it needs to be broadcast. I think that's correct...

Reply:
Also let's change that `contains?` into just `(:message/thread-broadcast? %)`, because if the value is `false` then `contains?` will still return true.